.github/workflows/ci.yaml

```diff
@@ -135,3 +135,27 @@ jobs:
           if [ "${cluster_status}" != "green" ]; then
               exit 1
           fi
+
+      - name: Upgrade snap
+        run: |
+          version="$(cat snap/snapcraft.yaml | yq .version)"
+          sudo snap install opensearch_${version}_amd64.snap --dangerous --jailmode
+
+          if [ "$(ls /var/snap/opensearch/x2)" ]; then
+              echo "Snap upgraded."
+          fi
+
+      - name: Ensure the cluster is reachable and node created after upgrade
+        run: |
+          # start opensearch after upgrade
+          sudo snap start opensearch.daemon
+          # Give some time for the service to come back up
+          sleep 100s
+
+          health_resp=$(sudo snap run opensearch.test-cluster-health-green)
+          echo -e "Cluster Health Response: \n ${health_resp}"
+          if [ "$(sudo snap run opensearch.test-cluster-health-green)" ]; then
+              echo -e "Cluster Health Response: \n ${health_resp}"
+          else
+              exit 1
+          fi
```

Comment:
We could perhaps simplify this with a single `if` when we're testing against the `status color` of the cluster like in previous tests